tests/unit/utils/validateRange.js

```diff
@@ -3,66 +3,71 @@ import assert from 'assert';
 import { parseRange } from
     '../../../lib/api/apiUtils/object/parseRange';
 
+function checkRange(rangeHeader, totalLength, expectedRange) {
+    const { range, error } =
+        parseRange(rangeHeader, totalLength);
+    assert.ifError(error);
+    assert.deepStrictEqual(range, expectedRange);
+}
+
 describe('parseRange function', () => {
     it('should return an array with the start and end if range is '
         + 'valid', () => {
-        const rangeHeader = 'bytes=0-9';
-        const totalLength = 10;
-        const actual =
-            parseRange(rangeHeader, totalLength);
-        assert.deepStrictEqual(actual, [0, 9]);
+        checkRange('bytes=0-9', 10, [0, 9]);
     });
 
     it('should set the end of the range at the total object length minus 1 ' +
         'if the provided end of range goes beyond the end of the object ' +
         'length', () => {
-        const rangeHeader = 'bytes=0-9';
-        const totalLength = 8;
-        const actual =
-            parseRange(rangeHeader, totalLength);
-        assert.deepStrictEqual(actual, [0, 7]);
+        checkRange('bytes=0-9', 8, [0, 7]);
     });
 
     it('should handle incomplete range specifier where only end offset is ' +
     'provided', () => {
-        const rangeHeader = 'bytes=-500';
-        const totalLength = 10000;
-        const actual = parseRange(rangeHeader, totalLength);
-        assert.deepStrictEqual(actual, [9500, 9999]);
+        checkRange('bytes=-500', 10000, [9500, 9999]);
     });
 
     it('should handle incomplete range specifier where only start ' +
     'provided', () => {
-        const rangeHeader = 'bytes=9500-';
-        const totalLength = 10000;
-        const actual = parseRange(rangeHeader, totalLength);
-        assert.deepStrictEqual(actual, [9500, 9999]);
+        checkRange('bytes=9500-', 10000, [9500, 9999]);
+    });
+
+    it('should return undefined for the range if the range header ' +
+        'format is invalid: missing equal', () => {
+        checkRange('bytes0-9', 10, undefined);
     });
 
     it('should return undefined for the range if the range header ' +
-        'format is invalid', () => {
-        const rangeHeaderMissingEquals = 'bytes0-9';
+        'format is invalid: missing dash', () => {
+        checkRange('bytes=09', 10, undefined);
+    });
+
+    it('should return undefined for the range if the range header ' +
+        'format is invalid: value invalid character', () => {
+        checkRange('bytes=%-4', 10, undefined);
+    });
+
+    it('should return undefined for the range if the range header ' +
+    'format is invalid: value not int', () => {
+        checkRange('bytes=4-a', 10, undefined);
```

Comment:
Minor, since it's undefined, you can just ignore that instead of specifying it explcitly.